### ISSUE TITLE
TabRouter throwing 'addListener' is not a function

### ISSUE THREAD

**Issue by prodigynelson** [react-navigation/react-navigation#3437]
### Current Behavior

I recently upgraded my react-native version, and was also advised by [expo](https://blog.expo.io/expo-sdk-v23-0-0-is-now-available-be0a8c655414) to upgrade react-navigation to ^1.0.0-beta.19. I am using a nested TabRouter to handle some navigation, and before the upgrade it worked fine, but now I am getting this issue: [screenshot](http://prntscr.com/iakvyu). I've traced the error to where my custom TabRouter component is created (note, this is the setup I borrowed from the [old version of reactnavigation's docs](https://github.com/react-navigation/react-navigation/blob/master/examples/NavigationPlayground/js/CustomTabs.js), which they've since updated)

    const CustomTabView = ({ router, navigation }) => {
        const { routes, index } = navigation.state;
        console.log('router', router)
        const ActiveScreen = router.getComponentForRouteName(routes[index].routeName);
        const navigationhelp = addNavigationHelpers({
            dispatch: navigation.dispatch,
            state: routes[index],
            navigation: navigation
        })
        return (
            <View style={{flex: 1}}>
                <ActiveScreen navigation={navigationhelp} /> (this is line 88)
            </View>
        );
    };

    export default createNavigationContainer(
        createNavigator(HomeTabNavigator)(CustomTabView)
    )
The issue lies on "navigation={navigationhelp}" because I am adding navigationhelpers to the active screen, and it doesn't like it at all, if I comment that out it works fine. It worked 100% fine before the upgrade to the latest beta, what gives? 

### Expected Behavior

My nested TabRouter to work with addedNavigationHelpers because I can't navigate from within those screens otherwise

### How to reproduce

Create a nested TabRouter and pass it navigation helpers

### Your Environment

"react-navigation": "^1.0.0-beta.19",
"react-native": "https://github.com/expo/react-native/archive/sdk-23.0.0.tar.gz",

**Comment by brentvatne** [comment]
see: https://github.com/react-navigation/react-navigation/blob/master/examples/NavigationPlayground/js/CustomTabs.js

let me know if code here resolves the issue

**Comment by prodigynelson** [comment]
@brentvatne that's exactly where I got the code to begin with when I implemented this ~2 months ago, and since upgrading to the latest version it doesn't work. Is there another method besides addNavigationHelpers that I can pass to my screen?

**Comment by brentvatne** [comment]
are you using redux? https://reactnavigation.org/docs/redux-integration.html

**Comment by prodigynelson** [comment]
@brentvatne adding the listener worked, thanks!

**Comment by prodigynelson** [comment]
Just for anybody that will read this afterwards; I tried following the [redux integration](https://reactnavigation.org/docs/redux-integration.html) that brent linked above, and because I have separate files that create my store, app navigator, and redux, I couldn't create an addListener const in the file I had the navigation. Instead, I had to export it from the store.js file and import it in my navigator file.
 
Additionally, my navigation reducer no longer works, I'll update later if I figure out how to re-implement it.
